### PR TITLE
[CN-663] Move logs from NewRelic to Grafana Cloud

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -184,21 +184,22 @@ jobs:
...
-          kubectl create namespace newrelic
-          helm install --namespace=newrelic newrelic-bundle -f .github/newrelic/newrelic-values.yaml \
-            --set "global.cluster=kind-operator" \
-            --set "global.licenseKey=${{ secrets.NEW_RELIC_LICENCE_KEY }}" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[0].name=K_CLUSTER_NAME,newrelic-logging.fluentBit.additionalEnvVariables[0].value=${{ env.CLUSTER_NAME }}" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[1].name=GITHUB_RUN_ID,newrelic-logging.fluentBit.additionalEnvVariables[1].value=\"${{ github.run_id }}\"" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[2].name=GITHUB_RUN_NUMBER,newrelic-logging.fluentBit.additionalEnvVariables[2].value=\"${{ github.run_number }}\"" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[3].name=GITHUB_SHA,newrelic-logging.fluentBit.additionalEnvVariables[3].value=${{ github.sha }}" \
-            newrelic/nri-bundle
+          kubectl create namespace ${NAMESPACE}
```
Comment (mtyazici) [inline .github/workflows/pull-request.yml:200]: We are deploying Grafana and metallb to the same namespace as the operator. Should we deploy them to another namespace to not collect their logs?

File: .github/workflows/nightly-wan-e2e-gke.yaml [hunk 2]
```diff
@@ -71,23 +71,23 @@ jobs:
...
       matrix:
-        cluster: ["src", "trg"]
+        cluster: [ "src", "trg" ]
     with:
       cluster_name: ${{ matrix.cluster }}-wan-cluster-${{ github.run_number }}
-      nr_cluster_name: wan-operator
+      cluster_type: gke-wan
       gh_run_id: ${{ github.run_id }}
       gh_run_number: ${{ github.run_number }}
       gh_sha: ${{ github.sha }}
```
Comment (mtyazici) [inline .github/workflows/nightly-wan-e2e-gke.yaml:86]: Should we give `namespaces` variable as input?

File: .github/workflows/nightly-ph-gke.yaml [hunk 3]
```diff
@@ -37,13 +37,13 @@ jobs:
...
+    needs: [ create-gke-cluster ]
+    uses: ./.github/workflows/grafana-agent.yaml
     secrets: inherit
     with:
       cluster_name: ${{ needs.create-gke-cluster.outputs.CLUSTER_NAME }}
-      nr_cluster_name: phone-home-operator
+      cluster_type: gke-phone-home
       gh_run_id: ${{ github.run_id }}
       gh_run_number: ${{ github.run_number }}
       gh_sha: ${{ github.sha }}
```
Comment (mtyazici) [inline .github/workflows/nightly-ph-gke.yaml:49]: Should we give `namespaces` variable as input?

File: .github/workflows/e2e-ocp.yaml [hunk 4]
```diff
@@ -49,34 +60,18 @@ jobs:
...
-          helm install --namespace=newrelic newrelic-bundle -f .github/newrelic/newrelic-values.yaml \
-            --set "global.cluster=${{ env.NR_CLUSTER_NAME }}" \
-            --set "global.licenseKey=${{ secrets.NEW_RELIC_LICENCE_KEY }}" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[0].name=K_CLUSTER_NAME,newrelic-logging.fluentBit.additionalEnvVariables[0].value=${{ env.CLUSTER_NAME }}" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[1].name=GITHUB_RUN_ID,newrelic-logging.fluentBit.additionalEnvVariables[1].value=\"${{ github.run_id }}\"" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[2].name=GITHUB_RUN_NUMBER,newrelic-logging.fluentBit.additionalEnvVariables[2].value=\"${{ github.run_number }}\"" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[3].name=GITHUB_SHA,newrelic-logging.fluentBit.additionalEnvVariables[3].value=${{ github.sha }}" \
-            --set "newrelic-logging.fluentBit.additionalEnvVariables[4].name=LOG_PARSER,newrelic-logging.fluentBit.additionalEnvVariables[4].value=cri" \
-            newrelic/nri-bundle
+          kubectl create namespace $GRAFANA_NS
```
Comment (mtyazici) [inline .github/workflows/e2e-ocp.yaml:65]: If somehow the namespace exists the tests will fail, I think we can keep the `| echo "namespace exists"` command

File: .github/workflows/e2e-eks.yaml [hunk 5]
```diff
@@ -69,22 +79,11 @@ jobs:
         run: |-
           aws eks update-kubeconfig --name "${{ env.CLUSTER_NAME }}"
 
-      - name: Install New Relic repo
+      - name: Install Grafana Agent agent
```
Comment (mtyazici) [inline .github/workflows/e2e-eks.yaml:82]:
```suggestion
      - name: Install Grafana Agent
```

**Review by mtyazici — COMMENTED** [review]
I left some comments, otherwise I can see logs in Grafana console. Thanks a lot :100:

File: .github/grafana/agent-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                  target_label: namespace
+                - action: replace
+                  source_labels:
+                    - __meta_kubernetes_pod_name
+                  target_label: pod
+                - action: replace
+                  source_labels:
+                    - __meta_kubernetes_pod_container_name
+                  target_label: container
+                - replacement: /var/log/pods/*$1/*.log
```
[the remark this answers is not in the document]
Comment (SeriyBg) [reply .github/grafana/agent-configmap.yaml:58]: The `$1` param won't be changed with the `envsubst`; it's the syntax for the Grafana agent itself and represents the parameter from the label match, in this case `__meta_kubernetes_pod_uid`

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (mtyazici) [reply .github/workflows/pull-request.yml:200]: But how the Grafana logs does not matter? Do they not log anything?

File: config/manager/kustomization.yaml [hunk 7]
```diff
@@ -9,6 +9,9 @@ images:
   newName: hazelcast/hazelcast-platform-operator
   newTag: latest-snapshot
 patches:
+- path: enable_developer_mode.yaml
+  target:
+    kind: Deployment
```
[the remark this answers is not in the document]
Comment (SeriyBg) [reply config/manager/kustomization.yaml]: Thanks for the catch, I've removed it manually as it requires regenerating bundle.yaml

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (SeriyBg) [reply .github/workflows/pull-request.yml:200]: Oh, you mean not to collect Grafana Agent logs? Yeah, you are right, will move it to another `ns`

Comment (mtyazici) [reply .github/workflows/pull-request.yml:200]: Yes, :+1:

**Review by hasancelik — APPROVED** [review]

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (SeriyBg) [reply .github/workflows/pull-request.yml]: Done ✅

**Review by mtyazici — APPROVED** [review]
LGTM :+1: 
Only problem left seems to be when multiple e2e tests are running on the OCP cluster they will have colliding grafana deployments. 
I also noticed that in the kind cluster we are deploying the metallb to the same namespace as the operator. Should we deploy metallb in `GRAFANA_NS`?

**Comment by SeriyBg** [comment]
@mtyazici, you are right regarding the OCP! I think we can deploy it in unique namespaces to avoid this problem.
Regarding the `metallb`, it seems that it is, in fact, deployed to `metallb-system` namespaces and not in the same as namespace with the operator

**Comment by mtyazici** [comment]
@SeriyBg, aah you are right. I was confused by the command `kubectl config set-context --current --namespace=$NAMESPACE`. I thought, metallb was deployed in the `$NAMESPACE`.

**Pull request merged**